cmd/auth-provider-gcp/plugin/plugin_test.go

```diff
@@ -0,0 +1,213 @@
+/*
```

Comment:
A couple test cases that come to mind (Okay to address in a follow up PR):
- metadata server returns error (e.g. 500 internal error)
- metadata server returns a response in an unexpected format
- metadata server takes too long to respond
-

Reply:
Yeah, I definitely think that would be good - since this is ideally done by tonight (maybe Friday at the latest), I'll see if I have some extra time left over to do these after addressing other feedback.